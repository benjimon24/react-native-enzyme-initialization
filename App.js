import React from "react";
import { StyleSheet, Text, View } from "react-native";

export default class App extends React.Component {
  render() {
    const setupFiles = `\"setupFiles\": \[
      \"<rootDir>/src/tempPolyfills.js\"\,
      \"<rootDir>/src/setupTests.js\"
    \]`;

    const scripts = `"test": "jest --verbose --watchAll"`;

    return (
      <View style={styles.container}>
        <Text>npm i --save-dev enzyme enzyme-adapter-react-16 react-native-mock</Text>
        <Text>npm install react-dom</Text>
        <Text>Add the following to your package.json:</Text>
        <Text> {setupFiles}</Text>
        <Text>Add the following parameters to scripts/test in your package.json:</Text>
        <Text> {scripts}</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    marginLeft: 20,
    flex: 1,
    backgroundColor: "#fff",
    justifyContent: "center"
  }
});
