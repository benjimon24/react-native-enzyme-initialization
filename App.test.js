import React from "react";
import App from "./App";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

describe("initialization", () => {
  const app = shallow(<App />);

  it("renders correctly", () => {
    expect(app).toMatchSnapshot();
  });

  it("renders without crashing", () => {
    const rendered = renderer.create(<App />).toJSON();
    expect(rendered).toBeTruthy();
  });
});
