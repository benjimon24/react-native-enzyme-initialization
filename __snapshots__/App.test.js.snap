// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initialization renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          npm i --save-dev enzyme enzyme-adapter-react-16 react-native-mock
        </Text>,
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          npm install react-dom
        </Text>,
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Add the following to your package.json:
        </Text>,
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
           
          "setupFiles": [
      "&lt;rootDir&gt;/src/tempPolyfills.js",
      "&lt;rootDir&gt;/src/setupTests.js"
    ]
        </Text>,
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Add the following parameters to scripts/test in your package.json:
        </Text>,
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
           
          "test": "jest --verbose --watchAll"
        </Text>,
      ],
      "style": Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "justifyContent": "center",
        "marginLeft": 20,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "npm i --save-dev enzyme enzyme-adapter-react-16 react-native-mock",
          "ellipsizeMode": "tail",
        },
        "ref": null,
        "rendered": "npm i --save-dev enzyme enzyme-adapter-react-16 react-native-mock",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "npm install react-dom",
          "ellipsizeMode": "tail",
        },
        "ref": null,
        "rendered": "npm install react-dom",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Add the following to your package.json:",
          "ellipsizeMode": "tail",
        },
        "ref": null,
        "rendered": "Add the following to your package.json:",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": Array [
            " ",
            "\\"setupFiles\\": [
      \\"<rootDir>/src/tempPolyfills.js\\",
      \\"<rootDir>/src/setupTests.js\\"
    ]",
          ],
          "ellipsizeMode": "tail",
        },
        "ref": null,
        "rendered": Array [
          " ",
          "\\"setupFiles\\": [
      \\"<rootDir>/src/tempPolyfills.js\\",
      \\"<rootDir>/src/setupTests.js\\"
    ]",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Add the following parameters to scripts/test in your package.json:",
          "ellipsizeMode": "tail",
        },
        "ref": null,
        "rendered": "Add the following parameters to scripts/test in your package.json:",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": Array [
            " ",
            "\\"test\\": \\"jest --verbose --watchAll\\"",
          ],
          "ellipsizeMode": "tail",
        },
        "ref": null,
        "rendered": Array [
          " ",
          "\\"test\\": \\"jest --verbose --watchAll\\"",
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            npm i --save-dev enzyme enzyme-adapter-react-16 react-native-mock
          </Text>,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            npm install react-dom
          </Text>,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            Add the following to your package.json:
          </Text>,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
             
            "setupFiles": [
      "&lt;rootDir&gt;/src/tempPolyfills.js",
      "&lt;rootDir&gt;/src/setupTests.js"
    ]
          </Text>,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            Add the following parameters to scripts/test in your package.json:
          </Text>,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
             
            "test": "jest --verbose --watchAll"
          </Text>,
        ],
        "style": Object {
          "backgroundColor": "#fff",
          "flex": 1,
          "justifyContent": "center",
          "marginLeft": 20,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "npm i --save-dev enzyme enzyme-adapter-react-16 react-native-mock",
            "ellipsizeMode": "tail",
          },
          "ref": null,
          "rendered": "npm i --save-dev enzyme enzyme-adapter-react-16 react-native-mock",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "npm install react-dom",
            "ellipsizeMode": "tail",
          },
          "ref": null,
          "rendered": "npm install react-dom",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Add the following to your package.json:",
            "ellipsizeMode": "tail",
          },
          "ref": null,
          "rendered": "Add the following to your package.json:",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": Array [
              " ",
              "\\"setupFiles\\": [
      \\"<rootDir>/src/tempPolyfills.js\\",
      \\"<rootDir>/src/setupTests.js\\"
    ]",
            ],
            "ellipsizeMode": "tail",
          },
          "ref": null,
          "rendered": Array [
            " ",
            "\\"setupFiles\\": [
      \\"<rootDir>/src/tempPolyfills.js\\",
      \\"<rootDir>/src/setupTests.js\\"
    ]",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Add the following parameters to scripts/test in your package.json:",
            "ellipsizeMode": "tail",
          },
          "ref": null,
          "rendered": "Add the following parameters to scripts/test in your package.json:",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": Array [
              " ",
              "\\"test\\": \\"jest --verbose --watchAll\\"",
            ],
            "ellipsizeMode": "tail",
          },
          "ref": null,
          "rendered": Array [
            " ",
            "\\"test\\": \\"jest --verbose --watchAll\\"",
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
